caddyprom.go

```diff
@@ -70,11 +70,9 @@ func (m *Metrics) Provision(ctx caddy.Context) error {
 
 // UnmarshalCaddyfile -
 func (m *Metrics) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
-	// for d.Next() {
-	// 	if !d.Args(&m.Addr) {
-	// 		return d.ArgErr()
-	// 	}
-	// }
+	for d.Next() {
+		d.Args(&m.Addr)
```

Comment:
Why not return the error here (like in the commented-out part removed above)?

Reply:
If you return the error there it will require the argument to be always present and the change will break existing Caddyfiles. I haven't however tried other combinations (e.g., with multiple arguments to the prometheus directive).

Reply:
Can you try the other combinations to be certain? Ideally there'd be unit tests to verify all the combinations, but if you're not comfortable with it I can add that before this merges.

Reply:
Additional parameters are ignored, as already happened before this change.